### PR TITLE
MonadRec instance for ReaderT

### PR DESCRIPTION
**Pull request by ethul**
Adding a MonadRec instance

### CODE REVIEW

File: src/Control/Monad/Reader/Trans.purs [hunk 1]
```diff
@@ -96,3 +100,12 @@ instance monadWriterReaderT :: (Monad m, MonadWriter w m) => MonadWriter w (Read
 instance distributiveReaderT :: (Distributive g) => Distributive (ReaderT e g) where
   distribute a = ReaderT \e -> collect (flip runReaderT e) a
   collect f = distribute <<< map f
+
+instance monadRecReaderT :: (MonadRec m) => MonadRec (ReaderT r m) where
+  tailRecM k a = ReaderT \r -> tailRecM k' (Tuple a r)
```
Comment (paf31) [inline src/Control/Monad/Reader/Trans.purs]: I don't think you need to tuple up the state here. You can just pass in `r` as it doesn't change.

**Comment by ethul** [comment]
You're right. I've updated the instance. Thanks!

**Comment by paf31** [comment]
:+1: Thanks!

**Comment by ethul** [comment]
Thanks for the merge!

**Pull request merged**